lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Bitdefender.pm

```diff
@@ -0,0 +1,81 @@
+package GLPI::Agent::Task::Inventory::Linux::AntiVirus::Bitdefender;
+
+use strict;
+use warnings;
+
+use parent 'GLPI::Agent::Task::Inventory::Module';
+
+use UNIVERSAL::require;
+use Cpanel::JSON::XS;
+
+use GLPI::Agent::Tools;
+
+sub isEnabled {
+    return canRun('bduitool');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $antivirus = _getBitdefenderInfo(logger => $logger);
+    if ($antivirus) {
+        $inventory->addEntry(
+            section => 'ANTIVIRUS',
+            entry   => $antivirus
+        );
+
+        $logger->debug2("Added $antivirus->{name}" . ($antivirus->{version} ? " v$antivirus->{version}" : ""))
+            if $logger;
+    }
+}
+
+sub _getBitdefenderInfo {
+    my (%params) =  (
+        command => '/opt/bitdefender-security-tools/bin/bduitool get ps',
+        @_
+    );
+
+    my $bduitool_output = getAllLines(%params)
+        or return;
+
+    my $product_version = '';
+    my $engines_version = '';
+    my $enabled = '';
+    my $antimalware_status = '';
+    my $new_update_available = '';
+    my $new_security_content_available = '';
+
+    foreach my $line (split("\n", $bduitool_output)) {
+        if ($line =~ /^Product version:/) {
+            ($product_version = $line) =~ s/^Product version:\s+//;
+        } elsif ($line =~ /^Engines version:/) {
+            ($engines_version = $line) =~ s/^Engines version:\s+//;
+        } elsif ($line =~ /^Antimalware status:/) {
+            ($antimalware_status = $line) =~ s/^Antimalware status:\s+//;
+        } elsif ($line =~ /^New product update available:/) {
+            ($new_update_available = $line) =~ s/^New product update available:\s+//;
+        } elsif ($line =~ /^New security content available:/) {
+            ($new_security_content_available = $line) =~ s/^New security content available:\s+//;
+        }
```

Comment:
The logic here is too complex, use better something like:
```suggestion
        my ($key, $value) = $line =~ /^([^:]+):\s+(.*)$/
            or next;
        if ($key eq "Product version") {
            $product_version = $value;
        } elsif ($key eq "Engines version") {
            $engines_version = $value;
        } elsif ($key eq "Antimalware status") {
            $antimalware_status = $value;
        } elsif ($key eq "New product update available") {
            $new_update_available = $value;
        } elsif ($key eq "New security content available") {
            $new_security_content_available = $value;
        }
```

Eventually, you can even avoid to use temporary variable and directly update the hash to return. The following could then replace l.44 to 80:
```
    my $av = {
        "name" => 'Bitdefender Endpoint Security Tools (BEST) for Linux',
        "manufacturer" => 'Bitdefender',
        "company" => 'Bitdefender',
        "uptodate" => 1,
    };

    foreach my $line (@output) {
        my ($key, $value) = $line =~ /^([^:]+):\s+(.+)$/
            or next;
        if ($key eq "Product version") {
            $av->{version} = $value;
        } elsif ($key eq "Engines version") {
            $av->{base_version} = $value;
        } elsif ($key eq "Antimalware status") {
            $av->{enabled} = $value eq "On" ? 1 : 0;
        } elsif ($key =~ /New (product update|security content) available/) {
            $av->{uptodate} = 0 if $value ne "no";
        }
    }

    return $av;
```

Reply:
This is too complex for my limited knowledge. I'll accept the suggestion :)